lib/internal/util/inspect.js

```diff
@@ -228,6 +231,33 @@ function getUserOptions(ctx) {
     getters: ctx.getters,
     ...ctx.userOptions
   };
+
+  // Typically, the target value will be an instance of `Object`. If that is
+  // *not* the case, the object may come from another vm.Context, and we want
+  // to avoid passing it objects from this Context in that case, so we remove
+  // the prototype from the returned object itself + the `stylize()` function,
+  // and remove all other non-primitives, including non-primitive user options.
+  if (isCrossContext) {
+    ObjectSetPrototypeOf(ret, null);
+    for (const key of ObjectKeys(ret)) {
+      if ((typeof ret[key] === 'object' || typeof ret[key] === 'function') &&
+          ret[key] !== null) {
+        delete ret[key];
+      }
+    }
+    ret.stylize = ObjectSetPrototypeOf((value, flavour) => {
+      let stylized;
+      try {
+        stylized = ctx.stylize(value, flavour);
+      } catch {}
+
+      if (typeof stylized !== 'string') return value;
```

Comment:
Is this required? The current stylize function does not check for strings and I would keep those aligned (especially, since it's not something supported to do).

Reply:
Isn’t the purpose of `.stylize()` to transform strings into other strings?

Reply:
Yes but this function is not filtering anything in the "regular context". I would try to keep the behavior aligned (and it's not a supported function anyway).

Reply:
> Yes but this function is not filtering anything in the "regular context". I would try to keep the behavior aligned

I can do this if you feel strongly about it, but imo the function is broken anyway if it does not return a string.

> and it's not a supported function anyway.

It’s used in examples in the documentation, so I think it qualifies as officially supported.

Reply:
(Also, practically speaking, I don’t think anybody is going to *override* this function with a custom one.)

Reply:
> It’s used in examples in the documentation, so I think it qualifies as officially supported.

Thanks for pointing that out (even though there's no description of what it would do). I was not aware of that.

> practically speaking, I don’t think anybody is going to override this function with a custom one.

The guard here would never be triggered with the internal stylize function.

> the function is broken anyway if it does not return a string.

Absolutely. It would either trigger an error in such cases or it changes the printed output to the stringified value. It would therefore align the behavior more closely by using my suggestion.

Reply:
> > practically speaking, I don’t think anybody is going to override this function with a custom one.
> 
> The guard here would never be triggered with the internal stylize function.

See the example above.

> > the function is broken anyway if it does not return a string.
> 
> Absolutely. It would either trigger an error in such cases or it changes the printed output to the stringified value. It would therefore align the behavior more closely by using my suggestion.

I can add stringification here as well, inside the try/catch. If I understand correctly, that is your main concern? I don’t see any issue with that.

Reply:
Okay, I’ve implemented that.